src/vmm/src/device_manager/persist.rs

```diff
@@ -223,26 +240,22 @@ impl<'a> Persist<'a> for MMIODeviceManager {
                 .map_err(Error::Net)?,
             ));
 
-            let device_id = net_state.device_id.clone();
-            let transport_state = &net_state.transport_state;
-            let mmio_slot = &net_state.mmio_slot;
-            dev_manager
-                .slot_sanity_check(mmio_slot)
-                .map_err(Error::DeviceManager)?;
-
-            let restore_args = MmioTransportConstructorArgs {
-                mem: mem.clone(),
-                device: device.clone(),
-            };
-            let mmio_transport = MmioTransport::restore(restore_args, transport_state)
-                .map_err(|()| Error::MmioTransport)?;
-            dev_manager
-                .register_virtio_mmio_device(vm, device_id, mmio_transport, &mmio_slot)
-                .map_err(Error::DeviceManager)?;
-
-            event_manager
-                .add_subscriber(device)
-                .map_err(Error::EventManager)?;
+            restore_helper(
+                device.clone(),
+                device.clone(),
+                &net_state.device_id,
+                &net_state.transport_state,
+                &net_state.mmio_slot,
+                constructor_args.event_manager,
+            )?;
+
+            // If device is activated, kick the net queue(s) to make up for any
+            // pending events we may have not captured in snapshot.
+            // Ratelimiters are restored 'unblocked' so nothing to do there.
```

Comment:
I guess we can be more explicit here and talk about inflight epoll events.

Reply:
done